### PR TITLE
Add dns server tests

### PR DESCRIPTION
**Pull request by johntdyer**

### CODE REVIEW

File: dns_server.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	MtuMeasurements       int            `json:"mtuMeasurements,omitempty"`
+	NetworkMeasurements   int            `json:"networkMeasurements,omitempty"`
+	NumPathTraces         int            `json:"numPathTraces,omitempty"`
+	ProbeMode             string         `json:"probeMode,omitempty"`
+	Protocol              string         `json:"protocol,omitempty"`
+	RecursiveQueries      int            `json:"recursiveQueries,omitempty"`
+}
+
+// AddDNSServer - Add dns server test
+func (t *DNSServer) AddDNSServer(id int) {
```
Comment (william20111) [inline dns_server.go:46]: Is this for adding agents or servers?

File: dns_server.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to delete page load, response code %d", resp.StatusCode)
+	}
+	return nil
+}
+
+//UpdateDNSServer - - delete dns server test
```
Comment (william20111) [inline dns_server.go:92]: update comment

File: dns_server.go (hunk 1 above)
Comment (johntdyer) [reply dns_server.go:46]: So the type of test is actually called a "dns-server" test

Comment (william20111) [reply dns_server.go:46]: https://github.com/william20111/go-thousandeyes/blob/f1ebee8d1f881bd73833455c16f279735899cb67/page_load.go#L52-L55

Cool, just for my sanity this is what we have in the page_load. It adds agents into the test, looks like this is doing the same but with different func name.

Comment (johntdyer) [reply dns_server.go]: i think l understand now.  Just pushed a fix.

File: dns_server.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+//DeleteDNSServer - delete dns server test
+func (c *Client) DeleteDNSServer(id int) error {
+	resp, err := c.post(fmt.Sprintf("/tests/dns-server/%d/delete", id), nil, nil)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to delete page load, response code %d", resp.StatusCode)
```
Comment (william20111) [inline dns_server.go:87]: sorry to be a pita, last thing. Then lets merge this. just need to update log msg

Comment (johntdyer) [reply dns_server.go]: No worries man, thats what code reviews are for

**Pull request merged**